torch/_inductor/ir.py

```diff
@@ -3897,17 +3897,20 @@ def should_allocate(self):
         return False
 
     def get_mutation_names(self):
-        assert isinstance(self.layout, MutationLayout)
-        return (self.layout.target.get_name(),)
+        return [self.inputs[0].get_name()]
+
+    def get_unbacked_symbol_defs(self):
+        return {}
```

Comment:
Sets are non-deteriministic, so we should avoid them in return types.

Reply:
https://github.com/pytorch/pytorch/blame/3be0e1cd587ece8fa54a3a4da8ae68225b9cbb9b/torch/_inductor/ir.py#L2664

@ezyang defined this to be a set return, do you want me to update all of them to be lists?

Reply:
Yeah, or update all the usage to call `sorted`.

Reply:
Using sorted results in
```
TypeError: cannot determine truth value of Relational
```
There are a lot of set type of operations being applied on the symbols, so moving them to list would convert these operations from O(1) to O(n). I'm not sure what the rigth decision there is but I'll keep this as a follow up since I am only introducing empty sets on this PR.

Reply:
Sounds good.  If you don't plan to fix this soon please create an issue.

Reply:
I don't have a good solution in mind, if you an idea, i'm happy to implement it.
https://github.com/pytorch/pytorch/issues/113130